src/server/handlers/create.rs

```diff
@@ -22,14 +32,49 @@ pub struct CreateSubnetResponse {
 }
 
 /// The create subnet json rpc method handler.
-pub(crate) struct CreateSubnetHandler {}
+pub(crate) struct CreateSubnetHandler<T: JsonRpcClient> {
+    shared: Arc<SubnetManagerShared<T>>,
+}
+
+impl <T: JsonRpcClient> CreateSubnetHandler<T> {
+    pub(crate) fn new(shared: Arc<SubnetManagerShared<T>>) -> Self {
+        Self { shared }
+    }
+}
 
 #[async_trait]
-impl JsonRPCRequestHandler for CreateSubnetHandler {
+impl <T: JsonRpcClient + Send + Sync> JsonRPCRequestHandler for CreateSubnetHandler<T> {
     type Request = CreateSubnetParams;
     type Response = CreateSubnetResponse;
 
-    async fn handle(&self, _request: Self::Request) -> anyhow::Result<Self::Response> {
-        Ok(CreateSubnetResponse{ address: String::from("/root") })
+    async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
+        let parent = &request.parent;
+
+        if !self.shared.contains_subnet(parent) {
+            return Err(anyhow!("target parent subnet not found"));
+        }
+
+        // this is safe to unwrap as we ensure this key exists.
+        let bundle = self.shared.get(parent).unwrap();
```

Comment:
This is really confusing, I had to read the whole code several times to understand what was happening under the hood. I think the `SharedSubnetManager` can be significantly simplified. We can discuss sync if it is easier, but I would suggest:
- I would maybe call it `SubnetManagerPool` instead of shared.
- Having a `new_pool(subnets) that initializes the managers pool for all subnets. 
- We could have a `.get(subnetID)` method that returns the manager for the subnet from the pool (similar to how you use it).
- We should have a `pool_update` to update the manager when the config is updated. We can maybe lazily do this directly in `get`, i.e. to check if we have updates